Sofa/Component/SolidMechanics/FEM/Elastic/tests/TriangleFEMForceField_stepTest.cpp

```diff
@@ -0,0 +1,103 @@
+﻿/******************************************************************************
+*                 SOFA, Simulation Open-Framework Architecture                *
+*                    (c) 2006 INRIA, USTL, UJF, CNRS, MGH                     *
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#include <sofa/component/solidmechanics/testing/ForceFieldTestCreation.h>
+#include <sofa/component/solidmechanics/fem/elastic/TriangleFEMForceField.h>
+#include <sofa/component/topology/container/constant/MeshTopology.h>
+
+
+namespace sofa
+{
+
+template <typename _TriangleFEMForceField>
+struct TriangleFEMForceField_stepTest : public ForceField_test<_TriangleFEMForceField>
+{
+    typedef typename _TriangleFEMForceField::DataTypes DataTypes;
+
+    typedef typename _TriangleFEMForceField::VecCoord VecCoord;
+    typedef typename _TriangleFEMForceField::VecDeriv VecDeriv;
+    typedef typename _TriangleFEMForceField::Coord Coord;
+    typedef typename _TriangleFEMForceField::Deriv Deriv;
+    typedef typename Coord::value_type Real;
+
+    VecCoord x;
+    VecDeriv v,f;
+
+    TriangleFEMForceField_stepTest()
```

Comment:
For test scene implementation I'm always hesitating in using the abstraction or the raw c++ API. 
Using abstraction (simpleapi, python) make the compilation much more efficient.